main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -1252,9 +1252,8 @@ object Parser2 {
       if (at(TokenKind.BracketL)) {
         Type.parameters()
       }
-      if (eat(TokenKind.Equal)) {
-        Type.ttype()
-      }
+      expect(TokenKind.Equal)
+      Type.ttype()
```

Comment:
I think the idea of `if (eat(TokenKind.Equal))` is to avoid two parser errors, perhaps this is best of both worlds?
```scala
if (eat(TokenKind.Equal)) {
  Type.ttype()
} else {
  expect(TokenKind.Equal)
}
```